File: cursed_time/lib/src/balatro/deck.py
from enum import Enum
from dataclasses import dataclass
from typing import Optional

class Rank(Enum):
    """Card ranks"""
    Two = "2"
    Three = "3"
    Four = "4"
    Five = "5"
    Six = "6"
    Seven = "7"
    Eight = "8"
    Nine = "9"
    Ten = "10"
    Jack = "Jack"
    Queen = "Queen"
    King = "King"
    Ace = "Ace"

class Suit(Enum):
    """Card suits"""
    Spades = "Spades"
    Hearts = "Hearts"
    Clubs = "Clubs"
    Diamonds = "Diamonds"

@dataclass
class Card:
    """Playing card"""
    rank: Rank
    suit: Suit
    
    def chip_value(self) -> int:
        """Get base chip value for this card"""
        chip_map = {
            Rank.Two: 2, Rank.Three: 3, Rank.Four: 4, Rank.Five: 5,
            Rank.Six: 6, Rank.Seven: 7, Rank.Eight: 8, Rank.Nine: 9,
            Rank.Ten: 10, Rank.Jack: 10, Rank.Queen: 10, Rank.King: 10,
            Rank.Ace: 11
        }
        return chip_map[self.rank]

File: client/lib/src/balatro/deck.rs
use serde::{Deserialize, Serialize};
use crate::balatro_enum;


#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct Card {
    pub edition: Option<Edition>,
    pub enhancement: Option<Enhancement>,
    pub rank: Rank,
    pub suit: Suit,
    pub seal: Option<Seal>
}

#[derive(Serialize, Deserialize, Debug, Clone, Copy)]
#[serde(rename_all = "PascalCase")]
pub enum Suit {
    Spades,
    Hearts,
    Clubs,
    Diamonds
}

#[derive(Serialize, Deserialize, Debug, Clone, Copy)]
#[serde(rename_all = "PascalCase")]
pub enum Rank {
    #[serde(rename = "Ace")]
    Ace,
    #[serde(rename = "2")]
    Two,
    #[serde(rename = "3")]
    Three,
    #[serde(rename = "4")]
    Four,
    #[serde(rename = "5")]
    Five,
    #[serde(rename = "6")]
    Six,
    #[serde(rename = "7")]
    Seven,
    #[serde(rename = "8")]
    Eight,
    #[serde(rename = "9")]
    Nine,
    #[serde(rename = "10")]
    Ten,
    Jack,
    Queen,
    King
}

balatro_enum!(Enhancement {
    Wild = "m_wild",
    Glass = "m_glass",
    Bonus = "m_bonus",
    Mult = "m_mult",
    Lucky = "m_lucky",
    Steel = "m_steel",
    Stone = "m_stone",
    Gold = "m_gold"
});

balatro_enum!(Seal {
    Blue = "blue_seal",
    Red = "red_seal",
    Purple = "purple_seal",
    Gold = "gold_seal"
});

balatro_enum!(Edition {
    None = "e_base",
    Foil = "e_foil",
    Holographic = "e_holo",
    Polychrome = "e_polychrome"
});
